### ISSUE TITLE
Put Python algorithms in classes

### ISSUE THREAD

**Issue by guimspace** [guimspace/envelope-matrix#3]
Python functions should be defined in classes called Matrices and Envelope.

**Comment by dmallya** [comment]
Hello there!

I'm new to contributing to open source projects so please bear with me. 
Is this as simple as wrapping the python files with the python class syntax?
If so, ``solve.py`` needs to be part of class ``Matrices`` and ``wrap.py`` needs to be part of class ``Envelope`` right (just making sure)?

Thanks for your patience.

**Comment by guimspace** [comment]
Hi.

Python class is new to me, so if you have ideas and suggestions, feel free to comment them.
Yes, everything can be wrapped in the **Envelope** class, except for the functions `unwrapEnvelope()`, `solveForwardSubstitution()` and `solveBackwardSubstitution()`: they go in the **Matrices** class because they return a matrix and a vector.
I also believe the functions `evaluateSolution()` and `evaluateMatrices()` - from the `test/test.py` file - can be wrapped in the **Matrices** class, but that can wait.

**Comment by dmallya** [comment]
Sounds good, I'll wrap em' and send em' your way.

**Comment by dmallya** [comment]
They're ready, do I make a new branch and push it to that?

**Comment by guimspace** [comment]
Yes, that is it. Please make sure you added `Signed-off-by: name <email>` in the commit message.

**Comment by dmallya** [comment]
I cant push, access denied. Do you need to add me as a contributor or something first?

**Comment by guimspace** [comment]
Try [GitHub Docs - Creating a pull request](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request).

Let me know how it goes.

---
Edit:

This is the correct article: [Creating a pull request from a fork ](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork)

**Comment by dmallya** [comment]
Should've gone through. 👍 
Would be awesome if you let me know if that was what you needed or if im on the wrong track.

**Comment by guimspace** [comment]
Thank you for your PR. I'm reviewing it and there is a few points to solve.

1. Your classes are missing the method `__init__()` to initialize new instances of envelopes and matrices. Because of that, I think we can replace the method `initEnvelope()` with `__init__()` like this:

```
-    def initEnvelope(self):
-        return {
-            'diagg': [],
-            'enve': [],
-            'enveCol': [],
-            'enveLin': []
-        }
-        
+    def __init__(self):
+        self.diagg = []
+        self.enve =[]
+        self.enveCol = []
+        self.enveLin = []
```

This allows the other methods to access the defined lists. Let me know how far you pretend to go on this issue.

2. Your commit message is too long because you missed a break line. You need to amend it with the `git` command `git -s --amend` and remove the `Signed-off-by: name <email>` from the title.
3. The author's name of the commit is showing as **[your**. If you are the author of the commit, it is better if you use the same name for author and `Signed-off-by`.

By the way, this issue affects the next major release 2.x.x, so it will take a while to push it to the `main` branch.

**Comment by dmallya** [comment]
For some reason it looks like the commit is going through from another github account that I use, but I've set the correct account (TrojanCodes) which should be the sole author from now on.

Regarding ``1`` I can create init methods but will this in turn require the code to be refactored as well? How simple would the refactoring be?

Regarding ``2`` I have attempted to amend the commit message and it seems to have gone through.

Also I'm not quite sure what to add for the ``__init__()`` in ``Matrices.py`` since the methods themselves return what is required.

Thanks for your continued guidance.

Edit: Looks like the ``[your`` is because I havent manually set a ``user.name`` and I have rectified the issue.
Edit 2: ``__init__()`` in ``Envelope.py`` created as specified and pushed.
